### PR TITLE
Noise Components

### PR DESCRIPTION
**Pull request by Tirpitz93**
This adds some more advanced noise components.
I wanted to test some models with more organic movement so I created this. I did not create the underlying library.
I believe this approach is better than using a bitmap/2D lookup to generate a signal.

Depends on https://github.com/deerel/OpenSimplexNoise
It may not be the fastest computationally, but it was the fastest to get up and running, I am planning to add more noise components in the future, when I get a successful compile using another library.
I was considering FastNoise2, but I was having trouble compiling it using MinGW and hopsan.

This commit adds an initial 3 blocks:
 - 1D simplex noise, based on simulation time with various constants to tweak to requirements.
 - Variable 1D Simplex based on an explicit x input. Fewer constants to tweak, but more control over input.
 - Variable 4D Simplex block with X, Y, Z, W inputs and a single output signal.

 No Help files yet.
 1 Example

4D Simplex:
image

1D and 1D Variable Simplex with same configuration:
image

### CODE REVIEW

**Comment by Tirpitz93** [comment]
After a couple fixes I was able to compile on another machine after using ` --recurse-submodules`  when cloning.

File: NoiseLib/noiseSeed.hpp [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+
+            //Get data pointers
+            
+
+            //Read input variables
+//            Seed = (*mpSeed);
+
+            //Initialization code
+            std::mt19937 gen (std::time(NULL));
```
Comment (robbr48) [inline NoiseLib/noiseSeed.hpp:97]: `std::time` has a resolution of whole seconds. If running a batch of simulations where each simulation takes less than one second, multiple simulation would get the same seed. 

Would it be possible to use `std::chrono::high_resolution_clock` instead, as is already done on some places in Hopsan? (requires `#include <chrono>`)
```
std::mt19937 gen(static_cast<long unsigned int>(std::chrono::high_resolution_clock::now().time_since_epoch().count()));```
```

File: NoiseLib/noiseSeed.hpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        void simulateOneTimestep() override
+        {
+            //Read input variables
+//            Seed = (*mpSeed);
+//
+//            //Simulation code
+//
+//
+//            //Write output variables
+//            (*mpSeed) = Seed;
```
Comment (robbr48) [inline NoiseLib/noiseSeed.hpp:114]: Commented code like this can be removed (also in other places)

**Review by robbr48 — COMMENTED** [review]
Looks good as far as I can see. I am no expert in simplex noise, but I assume the OpenSimplexNoise library is correct. My only comment is the initialization of the random seed component, see my other comment.

**Review by robbr48 — APPROVED** [review]
Looks good!

**Pull request merged**